crates/spirv-std/src/image.rs

```diff
@@ -136,6 +136,42 @@ impl<
         ACCESS_QUALIFIER,
     >
 {
+    // Note: #[inline] is needed because in vulkan, the component must be a constant expression.
```

Comment:
You could plausibly add a `const COMPONENT: u32` generic parameter, `#[rustc_legacy_const_generics(3)]`, and remove `component: u32` from the normal parameter list. AFAICT, this would result in the same call syntax working, but it gets automatically converted to the `const` generic.

(this is what `core::arch` uses for constant arguments - not sure how appropriate it is here, but as it stands, the argument isn't even required to be constant)

Reply:
Yeah, it's a bit of a bizarre case - when compiling for Vulkan, it's required to be constant, but when not, it's free to be a variable. I feel like there's just no good solution here.